101-bundle-json.md

```diff
@@ -852,13 +852,19 @@ The fields are defined as follows:
   - `EXTENSION NAME`: a unique name for an extension. Names SHOULD follow the dotted name format described earlier in this section.
   - The value of the extension must be valid JSON, but is otherwise undefined.
 
-The usage of extensions is undefined. However, bundles SHOULD be installable by runtimes that do not understand the extensions.
+### Required Extensions
+
+Some extensions MAY be required by a bundle and operations involving that bundle MUST fail if the runtime does not support that bundle. In this case, a bundle author MUST use the `requiredExtensions` object to define those extensions that are necessary for installing the bundle. The `requiredExtensions` object should contain a collection of key/value pairs. The key MUST be the `EXTENSION NAME` and the value MUST be a well known JSON Schema reference that can be used to validate the JSON data contained in the `custom` object.
+
+A runtime MUST validate that it supports any defined custom actions before performing any operation. However, bundles SHOULD be installable by runtimes that do not understand the extensions if they are NOT identified in the `requiredExtensions` field.  
```

Comment:
1. Weaken "MUST" to "SHOULD" for consistency with the above comment on allowing a user to "force" an operation.

2. Perhaps we should avoid the word "validate" here as it might imply just schema validation.

3. We should talk about "required" actions rather than "defined" actions.

4. Let's use "support" rather than "understand" for consistency.
```suggestion
A runtime SHOULD check that it supports any required custom actions before performing any operation. However, bundles SHOULD be installable by runtimes that do not support extensions that are NOT identified in the `requiredExtensions` field.  
```

Reply:
My bad, @glyn, related to my comment https://github.com/deislabs/cnab-spec/pull/220#discussion_r302651720. I intended in my comment that MUST should be applied to the validation step, but NOT to the `force` activity step. For example, I would suggest that a runtime MUST inform the user that it doesn't know how to perform the custom extension X, but if the user passes a `--force` argument that it will perform the action in any case because "the user knows best". 

If that makes sense to us all, how should we state this? Maybe something like, "A runtime MUST check that it supports any required custom action... and it MUST require an explicit override statement if the user insists on attempting to perform an action on the bundle." I'm open to wording here, as my goal is to break apart the MUST CHECK AND STOP portion from the SHOULD ENABLE OVERRIDE portion.